<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [5]:
# Membuat dataframes dari dictionary
import pandas as pd
dictionary = {u'2012-06-08': 388,
 u'2012-06-09': 388,
 u'2012-06-10': 388,
 u'2012-06-11': 389,
 u'2012-06-12': 389}

df = pd.DataFrame(dictionary.items(), columns=['Date', 'DateValue'])

print(df)


         Date  DateValue
0  2012-06-08        388
1  2012-06-09        388
2  2012-06-10        388
3  2012-06-11        389
4  2012-06-12        389


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [52]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd

df = pd.read_csv('diamonds-1.csv')  
df


Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawaban :

- 
    Dictionary : Array dengan index yang ditentukan yang berisi lebih dari satu setiap indexnya
                
    List       : Berisi dari satu item di setiap indexnya, seperti [1,2,3]
    
    Series     :  index Series dari 0 sampai n-1, atau bisa mengganti index selain angka, misal dengan alfabet. 
    
- Series hanya berisi satu list dengan index, sedangkan dataframe dapat dibuat lebih dari satu series (merupakan kumpulan dari data series)

Jawab: 

In [34]:
# code here

import pandas as pd
from pandas import Series

arr = Series([15, 35, 55, 75])
print(arr)


0    15
1    35
2    55
3    75
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [33]:
# code here
price_first_columns = df.price[0]
color_last_columns = df.color[53939]
print("baris pertama dari column price = " ,price_first_columns)
print("baris terakhir dari column color = " , color_last_columns)

baris pertama dari column price =  326
baris terakhir dari column color =  D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: metode-metode yang berkaitan dengan pengumpulan dan penyajian suatu gugus data sehingga memberikan informasi yang berguna

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [61]:
# code here
# Mengetahui deskriptif statistik
data_describe = df.describe()

#Rata-rata dari column price
print("Nilai rata-rata dari column price = ", data_describe['price']['mean'])


#nilai standar deviasi dari column depth
print("Nilai standar deviasi dari column depth = ", data_describe['depth']['std'])

#Berapa nilai maximum dari column carat
print("Nilai maximum dari column carat = ", data_describe['carat']['max'])

Nilai rata-rata dari column price =  3932.799721913237
Nilai standar deviasi dari column depth =  1.4326213188337733
Nilai maximum dari column carat =  5.01


Jawab:
Nilai rata-rata dari column price 3932.799721913237

Nilai standar deviasi dari column depth 1.4326213188337733

Nilai maximum dari column carat 5.01